### PR TITLE
Move trajectory dependency functions out of deprecated module

### PR DESCRIPTION
**Pull request by dkillick**
The interpolator in `iris.analysis.trajectory` relies on three non-deprecated functions in the deprecated module `iris.analysis._interpolate_private`. This moves these functions into the `trajectory` module.

### CODE REVIEW

File: lib/iris/analysis/trajectory.py [hunk 1]
```diff
@@ -27,14 +27,13 @@
 import math
 
 import numpy as np
+from scipy.spatial import cKDTree
 
-from cf_units import Unit
```
Comment (corinnebosley) [inline lib/iris/analysis/trajectory.py]: How come this has gone?

File: lib/iris/tests/unit/analysis/trajectory/test__nearest_neighbour_indices_ndcoords.py [hunk 2]
```diff
@@ -141,13 +140,9 @@ def test_sample_dictionary(self):
...
-        warn_call = self.patch(
-            'iris.analysis._interpolate_private.warn_deprecated')
-        result = nn_ndinds(cube, sample_point)
-        self.assertEqual(result, [(1, 2)])
-        self.assertEqual(warn_call.call_count, 1)
-        self.assertIn('dictionary to specify points is deprecated',
-                      warn_call.call_args[0][0])
+        exp_emsg = 'must be a list of \(coordinate, value\) pairs'
+        with self.assertRaisesRegexp(TypeError, exp_emsg):
+            nn_ndinds(cube, sample_point)
```
Comment (corinnebosley) [inline lib/iris/tests/unit/analysis/trajectory/test__nearest_neighbour_indices_ndcoords.py]: Is this the only test we have for nn_ind_ndcoords?

Comment (dkillick) [reply lib/iris/tests/unit/analysis/trajectory/test__nearest_neighbour_indices_ndcoords.py]: Well, there's this whole test module that I've moved... This was the only bit that actively needed updating though, as I've removed a [little bit of deprecated functionality](https://github.com/SciTools/iris/pull/2804/files#diff-372eebf3389e425b54bf8e165cc46d2eL220).

File: lib/iris/analysis/trajectory.py (hunk 1 above)
Comment (dkillick) [reply lib/iris/analysis/trajectory.py]: Unused import.

Comment (corinnebosley) [reply lib/iris/analysis/trajectory.py]: Okie dokie.

File: lib/iris/tests/unit/analysis/trajectory/test__nearest_neighbour_indices_ndcoords.py (hunk 2 above)
Comment (corinnebosley) [reply lib/iris/tests/unit/analysis/trajectory/test__nearest_neighbour_indices_ndcoords.py]: Yeah that makes sense.  Good.  I'm happy.

**Comment by corinnebosley** [comment]
@dkillick Nice.

**Comment by dkillick** [comment]
Woop! 🎉

**Pull request merged**